pandas/core/computation/eval.py

```diff
@@ -232,8 +237,13 @@ def eval(expr, parser='pandas', engine=None, truediv=True,
     pandas.DataFrame.query
     pandas.DataFrame.eval
     """
-    inplace = validate_bool_kwarg(inplace, 'inplace')
-    first_expr = True
+
+    inplace = validate_bool_kwarg(inplace, "inplace")
+    modifiable = hasattr(target, "__setitem__")
+
+    if inplace and not modifiable:
+        raise ValueError("Cannot modify the provided target inplace")
```

Comment:
For me the try excepts are not needed. When trying to set an object that does not support assigment with strings like a list, you already get a clear error message (and this is really a corner case, people should actually only pass dataframes)